### ISSUE TITLE
Race podium rewards for top3 

### ISSUE THREAD

**Issue by Zatline** [teeworlds/teeworlds#3027]
A while ago I was playing on a random Race server where the ranking *1 was holding a blue CTF flag for being the record holder on the server for the map that was played. In my opinion, a great little detail to make Race more rewarding for the most dedicated. 
My suggestion, therefore, is to have the top 3 receive plated race flags/ rocket launchers in the traditional sense of gold/silver/bronze.

As part of #2789 regarding "Effects, Graphics & Details"

![race_top3](https://user-images.githubusercontent.com/344608/151540195-2eb458a8-729d-4082-b888-cbebe122f4dc.png)![image2035](https://user-images.githubusercontent.com/344608/151541263-8c51404b-2b13-400c-bb0a-1b8d7fa7a8b3.png)

Sparkles white:
![sparkles](https://user-images.githubusercontent.com/344608/151541960-25183c3c-f1a2-4475-91b2-cca91735f9c4.png)
Sparkles for testing/ implementation (better visibility):
![sparkles_test](https://user-images.githubusercontent.com/344608/151541972-b8c7827b-49fd-429c-a028-495b4ecd5b9d.png)

The sparks themselves should rotate and grow/shrink similar to what other particles in the game do, the white circle ideally grows below the spark with a slight time offset, which should result in a nice little shiny animation. 
The sparks maybe should only be on the bright highlighted side of the flags and the brightest parts of the gun.

**Comment by jxsl13** [comment]
I wonder whether it has to be flags. It could be anything but coming up with some new entity might create other issues.

**Comment by hopetez** [comment]
I'm glad you like the idea of the flagholding tee as a reward, Zatline! [timakro](https://github.com/timakro) implemented this on the [unique-race](https://github.com/unique-clan/unique-race) servers ages ago to reward the best player on the map. 

> I wonder whether it has to be flags. It could be anything but coming up with some new entity might create other issues.

No, of course not! We have only made use of what was available to us with the game layer in 0.6. Some people made other graphics instead of the flag, one of them was a crown, which I thought was nice.

**Comment by Zatline** [comment]
> one of them was a crown, which I thought was nice.

I already have plans for a crown and even medals, all written down in a private document. The idea there basically was that the crown is only available for world record holders. However, that requires an actual world ranking system and an actual database that stores demo files and maps I imagine this could maybe run via teeworlds.com similar to what the game Track-Menia does, where players enter demo files onto the website to get ranked.

> I wonder whether it has to be flags.

Flags are the simplest thing I could think of since it is just a layer behind the tee and therefore doesn't require a lot of new code, and the code for that already exists, all that needs to be adjusted is the position of the flag to fit the image above.

**Comment by ChillerDragon** [comment]
> I already have plans for a crown and even medals written down in a private document. The idea there basically was that the crown is only available for world record holders.

Yes can we please have medal entities! That could finally be used as proper currency texture on servers that have one like city servers. None of the currently used textures look like coin and it is so unsatisfying to collect them.

**Comment by oy** [comment]
The identifier could be something else like shoes or hats etc. I don't think it has to be flags just because we already have them. If we find a better way to present the top3 we should go for it. Even if it's harder to implement.

**Comment by Zatline** [comment]
The problem with hats is that every skin has different shapes so it would lead to new design issues also I'd like to save that struggle for the implementation of an actual crown later that only the world record holder would get so hats will eventually get in the way of that. On the other hand, shoes seem way too small and insignificant, whatever the reward ends up being should be easy to see in the scoreboard and ingame while playing. That's why a race flag is perfect as it is the gametypes very icon, easy to implement, proven to look great, won't cause new issues while the plated guns are a cute little plus.

**Comment by oy** [comment]
Well servers could still use the ctf flags, so there might be an issue with a possible race flag.
Hats we already have - for xmas. But yeah, these 2 I mentioned were just examples we talked about in the past. The identifier could be anything. It's good to gather possible ideas. We might find something better.
Medals sound good too - there used to be sketches I think. They were in the old repository before we moved to github. Have to check if I still have it somewhere.

**Comment by AssassinTee** [comment]
I did a test implementation of the new podium features and I think it's really amazing, what do you think?
Only the sparkles are missing right now
